cms/models/placeholdermodel.py

```diff
@@ -299,4 +306,81 @@ def actions(self):
             self._actions_cache = getattr(field, 'actions', PlaceholderNoAction())
         return self._actions_cache
 
+    def get_cache_expiration(self, request, response_timestamp):
+        """
+        Returns the number of seconds (from «response_timestamp») that this
+        placeholder can be cached. This is derived from the plugins it contains.
+
+        This method must return: EXPIRE_NOW <= int <= MAX_EXPIRATION_IN_SECONDS
+
+        :type request: HTTPRequest
+        :type response_timestamp: datetime
+        :rtype: int
+        """
+        lang = getattr(request, 'lang', None)
+
+        min_ttl = MAX_EXPIRATION_TTL
+        for plugin_item in self.get_plugins(lang):
+            instance, plugin = plugin_item.get_plugin_instance()
```

Comment:
a placeholder with 20 plugins will result in 20 queries to the db because of this line.
I recommend we don't get the plugin instance unless it's **really** necessary.
This means, we can pass the plugin id and leave it to the plugin authors to get an instance if they needed for computing the expiration